medusa/cassandra_utils.py

```diff
@@ -183,10 +183,13 @@ def get_token(host_token_pair):
         return {
             self.hostname_resolver.resolve_fqdn(host.address): {
                 'tokens': tokens,
-                'is_up': host.is_up
+                'is_up': host.is_up,
+                'rack': host.rack,
+                'dc': host.datacenter
             }
             for host, tokens in host_tokens_pairs
-            if host.datacenter == datacenter
+            if host.datacenter == dc_rack_pair[0]
+            if host.rack == dc_rack_pair[1]
```

Comment:
This is restricting the tokenmap to contain only a single dc/rack pair. It's one of the problems (feature) we currently have with Medusa, which restricts all operations to the local DC.

I'd rather remove the `if host.rack == dc_rack_pair[1]` line for now, which will keep the existing behavior. We can evolve this behavior and give more flexibility in the future, as it's outside of the scope of this PR.

I gave a try at the current state of this PR and on a 6 nodes cluster spreading over 2 DCs and three racks, the backup-cluster command only backed up a single node. Removing the line will correctly backup all 3 nodes in the local DC.

Reply:
Sounds reasonable that we back this constraint off to support backwards compatibility.